docs/source/java/cdata.rst

```diff
@@ -0,0 +1,313 @@
+            <dependency>
+                <groupId>org.bytedeco</groupId>
+                <artifactId>javacpp</artifactId>
+                <version>1.5.7</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.arrow</groupId>
+                <artifactId>arrow-c-data</artifactId>
+                <version>7.0.0</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.arrow</groupId>
+                <artifactId>arrow-memory-netty</artifactId>
+                <version>7.0.0</version>
+            </dependency>
+        </dependencies>
+    </project>
+
+.. code-block:: java
+
+    import org.bytedeco.javacpp.annotation.Platform;
+    import org.bytedeco.javacpp.annotation.Properties;
+    import org.bytedeco.javacpp.tools.InfoMap;
+    import org.bytedeco.javacpp.tools.InfoMapper;
+
+    @Properties(
+            target = "CDataInterfaceLibrary",
+            value = @Platform(
+                    include = {"CDataInterfaceLibrary.h"},
+                    compiler = {"cpp11"},
+                    linkpath = {"/arrow/cpp/build/debug/"},
+                    link = {"arrow"}
+            )
+    )
+    public class CDataInterfaceLibraryConfig implements InfoMapper {
+        @Override
+        public void map(InfoMap infoMap) {
+        }
+    }
+
+.. code-block:: shell
+
+    // Compile our Java code
+    javac -cp javacpp-1.5.7.jar CDataInterfaceLibraryConfig.java
+
+    // Generate CDataInterfaceLibrary
+    java -jar javacpp-1.5.7.jar CDataInterfaceLibraryConfig.java
+
+    // Generate libjniCDataInterfaceLibrary.dylib
+    java -jar javacpp-1.5.7.jar CDataInterfaceLibrary.java
+
+    // Validate libjniCDataInterfaceLibrary.dylib created
+    otool -L macosx-x86_64/libjniCDataInterfaceLibrary.dylib
+    macosx-x86_64/libjniCDataInterfaceLibrary.dylib:
+            libjniCDataInterfaceLibrary.dylib (compatibility version 0.0.0, current version 0.0.0)
+            @rpath/libarrow.800.dylib (compatibility version 800.0.0, current version 800.0.0)
+            /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1200.3.0)
+            /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)
+
+**Java Test**
+
+Let's create a Java class to Test C Data Interface from Java to C++:
+
+.. code-block:: java
+
+    import org.apache.arrow.c.ArrowArray;
+    import org.apache.arrow.c.ArrowSchema;
+    import org.apache.arrow.c.Data;
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.BigIntVector;
+
+    public class TestCDataInterfaceV5 {
+
+        public static void main(String[] args) {
+            CDataInterfaceLibrary.ArrowSchema arrowSchema = new CDataInterfaceLibrary.ArrowSchema();
+            CDataInterfaceLibrary.export_int32_type(arrowSchema);
+
+            ArrowSchema arrow_schema = ArrowSchema.wrap(arrowSchema.address());
```

Comment:
I think in general, this example isn't structured correctly. The specification states that the consumer is expected to allocate the array. So what we should do is:

1. Use ArrowSchema.allocateNew to create the structure
2. Pass the address via javacpp to the JNI function, which populates the structure

We shouldn't be defining our own ArrowSchema class here, and we shouldn't be expecting the C++ code to allocate the ArrowSchema itself.

Reply:
Changed on this way, thank you